Gems/EditorPythonBindings/Code/Source/PythonSystemComponent.h

```diff
@@ -60,11 +61,13 @@ namespace EditorPythonBindings
         
     private:
         class SymbolLogHelper;
+        class PythonGILScopedLock;
 
         // handle multiple Python initializers and threads
         AZStd::atomic_int m_initalizeWaiterCount {0};
         AZStd::semaphore m_initalizeWaiter;
         AZStd::recursive_mutex m_lock;
+        int m_lockRecursiveCounter = 0;
```

Comment:
any use of an atomic_int here?

Reply:
No need to make it `atomic_int`, every access to `m_lockRecursiveCounter` is done once the mutex `m_lock` is locked.

Reply:
even if it's protected, I believe it can happen that the different cores will see different values on their cache if its not atomic. With atomic you can make sure that never happens